### PR TITLE
Server drive query information request

### PR DESCRIPTION
**Pull request by ibeckermayer**

### CODE REVIEW

File: lib/srv/desktop/rdp/rdpclient/src/rdpdr/filesystem.rs [hunk 1]
```diff
@@ -471,6 +494,104 @@ impl FilesystemBackend {
...
+                },
+            ));
+        }
+        let file = file.unwrap(); // safe because of the if statement above
+        let device_io_response =
+            efs::DeviceIoResponse::new(rdp_req.device_io_request.clone(), NtStatus::SUCCESS);
+
+        // We support all the FsInformationClasses that FreeRDP does here
+        // https://github.com/FreeRDP/FreeRDP/blob/511444a65e7aa2f537c5e531fa68157a50c1bd4d/channels/drive/client/drive_file.c#L482
+        match rdp_req.file_info_class_lvl {
```
Comment (probakowski) [inline lib/srv/desktop/rdp/rdpclient/src/rdpdr/filesystem.rs:519]: Can we extract branches below to separate methods? It's hard to follow

File: lib/srv/desktop/rdp/rdpclient/src/rdpdr/filesystem.rs [hunk 2]
```diff
@@ -471,6 +494,104 @@ impl FilesystemBackend {
...
+                efs::ClientDriveQueryInformationResponse {
+                    device_io_response: efs::DeviceIoResponse::new(
+                        rdp_req.device_io_request.clone(),
+                        NtStatus::UNSUCCESSFUL,
+                    ),
+                    buffer: None,
+                },
+            ));
+        }
+        let file = file.unwrap(); // safe because of the if statement above
```
Comment (probakowski) [inline lib/srv/desktop/rdp/rdpclient/src/rdpdr/filesystem.rs:513]: I'd rather avoid `unwrap`

```suggestion
        let file = match file {
            Some(file) => file,
            None => {
                return Ok(RdpdrPdu::ClientDriveQueryInformationResponse(
                    efs::ClientDriveQueryInformationResponse {
                        device_io_response: efs::DeviceIoResponse::new(
                            rdp_req.device_io_request.clone(),
                            NtStatus::UNSUCCESSFUL,
                        ),
                        buffer: None,
                    },
                ))
            }
        };
```

Comment (ibeckermayer) [reply lib/srv/desktop/rdp/rdpclient/src/rdpdr/filesystem.rs]: 5bd154eae4971e3e7668087a2594f8e0eff841f6

File: lib/srv/desktop/rdp/rdpclient/src/rdpdr/filesystem.rs (hunk 1 above)
Comment (ibeckermayer) [reply lib/srv/desktop/rdp/rdpclient/src/rdpdr/filesystem.rs:519]: This gets modified in https://github.com/gravitational/teleport/pull/34139:

https://github.com/gravitational/teleport/pull/34139/files#diff-1b8bd3bdf7e3fb4382d5b034f19725c768263110c7648f3199d74fc77558002fR517-R619

Moving this discussion there.

File: lib/srv/desktop/rdp/rdpclient/src/rdpdr/filesystem.rs (hunk 2 above)
Comment (ibeckermayer) [reply lib/srv/desktop/rdp/rdpclient/src/rdpdr/filesystem.rs:513]: This gets modified in https://github.com/gravitational/teleport/pull/34139:

https://github.com/gravitational/teleport/pull/34139/files#diff-1b8bd3bdf7e3fb4382d5b034f19725c768263110c7648f3199d74fc77558002fR517-R619

Moving this discussion there.

Comment (ibeckermayer) [reply lib/srv/desktop/rdp/rdpclient/src/rdpdr/filesystem.rs:513]: a59a44ede95a22ba2fa1c848c78668c87cf1d71d

File: lib/srv/desktop/rdp/rdpclient/src/rdpdr/filesystem.rs (hunk 1 above)
Comment (ibeckermayer) [reply lib/srv/desktop/rdp/rdpclient/src/rdpdr/filesystem.rs:519]: a59a44ede95a22ba2fa1c848c78668c87cf1d71d

**Review by probakowski — APPROVED** [review]
